crates/gen/src/parser/element_type.rs

```diff
@@ -110,24 +110,25 @@ impl ElementType {
                         ("System", "Guid") => Self::Guid,
                         ("Windows.Win32.System.Com", "IUnknown") => Self::IUnknown,
                         ("Windows.Foundation", "HResult") => Self::HRESULT,
-                        ("Windows.Win32.System.Com", "HRESULT") => Self::HRESULT,
-                        ("Windows.Win32.System.WinRT", "HSTRING") => Self::String,
                         ("Windows.Win32.System.WinRT", "IInspectable") => Self::IInspectable,
                         ("Windows.Win32.System.SystemServices", "LARGE_INTEGER") => Self::I64,
                         ("Windows.Win32.System.SystemServices", "ULARGE_INTEGER") => Self::U64,
                         ("Windows.Win32.Graphics.Direct2D", "D2D_MATRIX_3X2_F") => Self::Matrix3x2,
                         ("System", "Type") => Self::TypeName,
                         _ => type_ref.resolve().into(),
                     },
-                    TypeDefOrRef::TypeDef(type_def) =>
-                    // Need to "re-resolve" the TypeDef as it may point to an arch-specific
-                    // definition. This lets the TypeTree be built for a specific architecture
-                    // without accidentally pulling in the wrong definition.
-                    {
-                        TypeReader::get()
-                            .resolve_type_def(type_def.namespace(), type_def.name())
-                            .into()
-                    }
+                    TypeDefOrRef::TypeDef(type_def) => match type_def.full_name() {
```

Comment:
This is something I still need to fix. There's no guarantee the types will be TypeDefs or TypeRefs. I was hoping in practice they'd only be TypeRefs as that's simpler but as long as the TypeDef resides in the same PE file then they can be TypeDefs. Ideally the arch-specific types would be TyepRefs though. That's a bug in the metadata (see comment below).

Reply:
#845 deals with these issues.

Reply:
OK, will complete 845.

Reply:
Ah, by kicking off the second commit the metadata import from nuget got lost as well, apologies. Thanks for pushing that back @kennykerr :)